backend/src/prompts/templates/generate-cypher-query.j2

```diff
@@ -1,33 +1,52 @@
+{
+    "question": <question provided by the user>, 
     "query": <cypher query>
+}
+
+The value for "query" must strictly be a valid Cypher query and must not contain any characters outside of Cypher syntax.
 
-}.
+If you cannot make a query, "query" should just say "None".
 
-The value for "query" must strictly be a valid CYPHER query and not contain anything other characters, that are not part of a Cypher query. 
+**Requirements:**
 
-If you cannot make a query, query should just say "None"
 
-Only use relationships, nodes and properties that are present in the schema below. 
+1. **Schema Usage**: Only use relationships, nodes, and properties that are present in the schema provided below. You are NOT ALLOWED to create new relationships, nodes, or properties not listed in the graph schema.
 
-You are NOT ALLOWED to create new relationships, nodes or properties that do not exist in the graph schema, under any circumstances. 
+2. **Query Scope**: You are only able to make queries that retrieve information. Do not create, delete, or update any entries.
 
-You are only able to make queries that search for information, you are not able to create, or delete or update entries.
+3. **Strict Syntax**: Follow Cypher syntax rules. Avoid introducing variables within clauses that do not support them.
 
-You must strictly follow cypher syntax rules and you are NOT ALLOWED to introduce variables inside clauses that do not allow it. 
+4. **Aggregation Requirements**: If a task requires finding the highest or lowest values, your query should retrieve all entries tied at the top value rather than limiting to a single entry.
+    Example: If there are multiple funds with the highest ESG social score in a specific industry, return all of them.
 
-Expenses are recorded as negative numbers, therefore a larger negative number represents a higher expense.
+5. **Relational Path**:
+    - Ensure the relational path aligns with the schema for all queries.
+    - When querying for a category case-insensitively, use `=~ '(?i)...'`.
+    - Example: To find a fund related to the `Aviation` industry with a `Social` ESG score, use:
+    
+    ```plaintext
+    MATCH (f:Fund)-[:CONTAINS]->(c:Company)-[:BELONGS_IN_INDUSTRY]->(i:Industry), (c)-[:HAS_ESG_SCORE]->(esg:ESGScore)
+    WHERE i.Name =~ '(?i)Aviation' AND esg.Category =~ '(?i)Social'
+    RETURN ...
+    ```
+    
+6. **Property Matching**: Adhere to exact property values and capitalization in the schema (e.g., 'Aviation' and 'Social').
 
-For example, an expense of -45 is greater than an expense of -15.
+7. **Single Result for Maximum/Minimum**:
+    - For queries seeking a single result with the "highest" or "lowest" value, use `ORDER BY` and `LIMIT 1` to return only the top result.
+    - Example: If finding the fund with the highest ESG social score, sort by `esg.Score DESC` and limit to 1 result.
 
-When returning a value, always remove the `-` sign before the number.
+8. **Expense Handling**:
```

Comment:
Do we need still need this expense handling logic?

Reply:
We don't actually need this. I just labelled the whole validation prompt. There is a lot of legacy code related to InferGPT, so I believe we should create a Ticket later on to remove all that.